### ISSUE TITLE
Bug: Tests don't fail under special conditions when using ExpectedExceptionSupport

### ISSUE THREAD

**Issue by sealor** [junit-team/junit5#1075]
## Overview

**Bug report**

I noticed that my tests don't fail anymore when using `ExpectedExceptionSupport` without an exception rule. You can check that behaviour with the appended JUnit tests.

JUnit5 version:
```
<dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter-migrationsupport</artifactId>
    <version>5.0.0</version>
    <scope>test</scope>
</dependency>
```

### 1st Test: ExceptionSupportTest
Actual behaviour: test succeeds.
Expected behaviour: test fails.

```
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.junit.jupiter.migrationsupport.rules.ExpectedExceptionSupport;

@ExtendWith(ExpectedExceptionSupport.class)
public class ExceptionSupportTest {

    @Test
    public void shouldFail() {
        throw new RuntimeException();
    }

}
```

### 2nd Test: WorkingExceptionSupportTest
Works as expected.

```
import org.junit.Rule;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.junit.jupiter.migrationsupport.rules.ExpectedExceptionSupport;
import org.junit.rules.ExpectedException;

@ExtendWith(ExpectedExceptionSupport.class)
public class WorkingExceptionSupportTest {

    @Rule
    public final ExpectedException exception = ExpectedException.none();

    @Test
    public void failsAsExpected() {
        throw new RuntimeException();
    }

}
```
### 3rd Test: More usual test case
Actual behaviour: test succeeds.
Expected behaviour: test fails.

```
import org.junit.Rule;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.migrationsupport.rules.EnableRuleMigrationSupport;
import org.junit.rules.TemporaryFolder;

import java.io.IOException;

// hint: EnableRuleMigrationSupport also enables ExpectedExceptionSupport
@EnableRuleMigrationSupport
public class MigrationSupportTest {

    @Rule
    public TemporaryFolder folder = new TemporaryFolder();

    @Test
    public void shouldFail() throws IOException {
        // this illegal operation throws an exception but the test succeeds
        folder.newFolder("/\\");
    }

}
```

**Comment by sormuras** [comment]
Duplicate of https://github.com/junit-team/junit5/issues/1069 ?

**Comment by marcphilipp** [comment]
Duplicate of #1069. Will be fixed in 5.0.1. See https://github.com/junit-team/junit5/issues/1069#issuecomment-330273039 for workarounds.
